src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Cipher.java

```diff
@@ -814,13 +814,23 @@ private int implDoFinal(byte[] out, int outOfs, int outLen)
                 if (paddingObj != null) {
                     int startOff = 0;
                     if (reqBlockUpdates) {
-                        startOff = padBufferLen;
+                        // padding is added to data in padBuffer
+                        // unless padBuffer is already full in which case
+                        // EncryptUpdate is performed and whole new padding
+                        // block is created
```

Comment:
suggestion on comment: 
// call C_EncryptUpdate first if the padBuffer is full to make room for padding bytes